ntbs-service/Services/AuthorizationService.cs

```diff
@@ -0,0 +1,82 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using ntbs_service.Models;
+using ntbs_service.Models.Enums;
+
+namespace ntbs_service.Services
+{
+    public interface IAuthorizationService
+    {
+        Task<bool> CanEdit(ClaimsPrincipal user, Notification notification);
+        Task<IEnumerable<Notification>> FilterNotificationsByUserAsync(ClaimsPrincipal user, IEnumerable<Notification> notifications);
+    }
+
+    public class AuthorizationService : IAuthorizationService
+    {
+        private readonly IUserService userService;
+        private UserPermissionsFilter filter;
+
+        public AuthorizationService(IUserService userService)
+        {
+            this.userService = userService;
+        }
+
+        public async Task<bool> CanEdit(ClaimsPrincipal user, Notification notification)

```

Comment:
This method is a singleton instance of the `FilterNotificationsByUserAsync` - would making use of that commonality be helpful?

Reply:
Yes I should probably be combining these methods - while it's a bit weird to create a list of notifications from a single one just so I can call the other method, it's also weird to be duplicating the logic and more error-prone.